### PR TITLE
support for sparse input

### PR DESCRIPTION
**Pull request by hacktuarial**
addresses #2

### CODE REVIEW

File: ordreg/ordinal.py [hunk 1]
```diff
@@ -10,6 +11,10 @@ def __init__(self, C=0.0, random_state=None):
         self.random_state = random_state
 
     def fit(self, X, y):
+        X, y = check_X_y(X, y, accept_sparse=True, multi_output=True)
+        if sparse.issparse(y):
+            y = y.todense()
```
Comment (hacktuarial) [inline ordreg/ordinal.py]: I couldn't figure out how to make sparse y work, but it seems like it should be possible.

File: ordreg/ordinal.py [hunk 2]
```diff
@@ -33,7 +38,7 @@ def objective(params):
 
             mass = np.diff(dfunc, axis=1)
 
-            v = (- np.sum(np.log(np.sum(mass * y, axis=1))) +
+            v = (- np.sum(np.log(np.sum(np.multiply(mass, y), axis=1))) +
```
Comment (hacktuarial) [inline ordreg/ordinal.py]: I'm on numpy 1.14.3, Python 3.6.1; I think my system was treating `*` as matrix multiplication, when I think it's supposed to be element-wise multiplication.

Comment (dsaxton) [reply ordreg/ordinal.py]: I think that makes sense.  (Also forgive my clumsiness and / or slowness, I'm quite new to reviewing pull requests.)

File: test/test_sparse_input.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+@pytest.fixture
+def X_sparse(X_dense):
+    return sparse.csr_matrix(X_dense)
+
+
+@pytest.fixture
+def y_dense():
+    y = np.random.choice(list(range(J)), size=N)
+    y = pd.get_dummies(y).values
```
Comment (dsaxton) [inline test/test_sparse_input.py]: Can we create `y` using only `numpy`?

Comment (hacktuarial) [reply test/test_sparse_input.py]: good idea! will make the change now

**Comment by hacktuarial** [comment]
What do you think about using OneHotEncoder from sklearn? It's very convenient, but we can work around it.

**Comment by dsaxton** [comment]
I think using `OneHotEncoder` would be good as well (since the idea is to have something that is `sklearn` compatible anyways, most things from that package should be fair game in my opinion)

File: test/test_wine_example.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    df = pd.read_csv('test/data/wine.csv')
+    for x in ['temp', 'contact']:
+        df[x] = df[x].astype('category').cat.codes
+    df['interaction'] = df['temp'] * df['contact']
+    return df
+
+
+def test_wine(wine):
+    X = wine[['temp', 'contact', 'interaction']].values
+    y = pd.get_dummies(wine['rating']).values
```
Comment (dsaxton) [inline test/test_wine_example.py]: Perhaps use `OneHotEncoder` here as well?

**Comment by hacktuarial** [comment]
Sorry about that! I mistakenly rebased one of my branches off the other.

**Comment by dsaxton** [comment]
Thanks again!

**Pull request merged**